Go Client/Go Rules.cpp

```diff
@@ -1,93 +1,93 @@
+   {
+      if (NumberOflibertiesOfAGroup(x, y) == 0)
+      {
+         KillGroup(x, y, goBoard);
+      }
+   }
 }
 
 void Rules::KillGroup(uint8_t x, uint8_t y, GameState &goBoard)
 {
-	goBoard.IncrementDeadStones(currentStone);
-	goBoard.PlayStone(x, y, stone::empty);
-	currentBoard = goBoard.GoBoard();
-	if (x < currentBoard.size() - 1)
-		if (currentBoard[y][x + 1] == currentStone)
-			KillGroup( x + 1, y, goBoard);
-	if (y < currentBoard.size() - 1)
-		if (currentBoard[y + 1][x] == currentStone)
-			KillGroup( x, y + 1, goBoard);
-	if (x)
-		if (currentBoard[y][x - 1] == currentStone)
-			KillGroup( x - 1, y, goBoard);
-	if (y)
-		if (currentBoard[y - 1][x] == currentStone)
-			KillGroup( x, y - 1,  goBoard);
+   goBoard.IncrementDeadStones(m_currentStone);
+   goBoard.PlayStone(x, y, stone::empty);
+   m_currentBoard = goBoard.GoBoard();
+   if (x < m_currentBoard.size() - 1)
+      if (m_currentBoard[y][x + 1] == m_currentStone)
+         KillGroup(x + 1, y, goBoard);
+   if (y < m_currentBoard.size() - 1)
+      if (m_currentBoard[y + 1][x] == m_currentStone)
+         KillGroup(x, y + 1, goBoard);
+   if (x)
+      if (m_currentBoard[y][x - 1] == m_currentStone)
+         KillGroup(x - 1, y, goBoard);
+   if (y)
+      if (m_currentBoard[y - 1][x] == m_currentStone)
+         KillGroup(x, y - 1, goBoard);
 }
 
 uint8_t Rules::NumberOflibertiesOfAGroup(uint8_t x, uint8_t y)
 {
-	coordinate currentCoord;
-	currentCoord.x = x;
-	currentCoord.y = y;
+   coordinate currentCoord;
+   currentCoord.x = x;
+   currentCoord.y = y;
 
-	for (uint16_t i = 0; i < checkedCoordinates.size(); i++)
-	{
-		if (currentCoord.x == checkedCoordinates[i].x && currentCoord.y == checkedCoordinates[i].y)
-		return liberties;
-	}
-	checkedCoordinates.push_back(currentCoord);
+   for (uint16_t i = 0; i < m_checkedCoordinates.size(); i++)
+   {
+      if (currentCoord.x == m_checkedCoordinates[i].x && currentCoord.y == m_checkedCoordinates[i].y)
+         return m_liberties;
+   }
+   m_checkedCoordinates.push_back(currentCoord);
 
-	CheckLibertiesOfAStone(x, y);
-	if (x < currentBoard.size() - 1)
-		if (currentBoard[y][x + 1] == currentStone)
-			NumberOflibertiesOfAGroup(x + 1, y);
-	if (y < currentBoard.size() - 1)
-		if (currentBoard[y + 1][x] == currentStone)
-			NumberOflibertiesOfAGroup(x, y + 1);
-	if (x)
-		if (currentBoard[y][x - 1] == currentStone)
-			NumberOflibertiesOfAGroup(x - 1, y);
-	if (y)
-		if (currentBoard[y - 1][x] == currentStone)
-			NumberOflibertiesOfAGroup(x, y - 1);
-	return liberties;
+   CheckLibertiesOfAStone(x, y);
+   if (x < m_currentBoard.size() - 1)
+      if (m_currentBoard[y][x + 1] == m_currentStone)
+         NumberOflibertiesOfAGroup(x + 1, y);
+   if (y < m_currentBoard.size() - 1)
+      if (m_currentBoard[y + 1][x] == m_currentStone)
+         NumberOflibertiesOfAGroup(x, y + 1);
+   if (x)
+      if (m_currentBoard[y][x - 1] == m_currentStone)
+         NumberOflibertiesOfAGroup(x - 1, y);
+   if (y)
```

Comment:
it does return the number of liberties of a group but also increments m_leberties... ummm yeah CheckNumberOfLibertiesOfAGroup is better tho :)